proto/scheduler.proto

```diff
@@ -116,6 +116,140 @@ message TaskSize {
   repeated CustomResource custom_resources = 4;
 }
 
+// CgroupSettings defines Linux cgroup2 options for an execution.
+//
+// Where applicable, the value -1 means "unlimited", which translates to the
+// "max" string in cgroup2 terms.
+//
+// Note: for numeric fields, a corresponding field prefixed with "use_" is
+// defined, which indicates whether the executor should apply the value. If it's
+// false, the setting should not be applied, meaning the default value is used
+// instead. This convention is used instead of defining 0 to mean "do not set,"
+// because in some cases it's valid to set a value to 0. proto3 optional is not
+// used because of the awkward Go API.
+message CgroupSettings {
+  // Proportion of CPU given to this task relative to other tasks in the parent
+  // cgroup. This provides for a best-effort CPU guarantee.
+  //
+  // Values 1 to 10000 are supported.
+  //
+  // Maps to "cpu.weight" in cgroup2.
+  int64 cpu_weight = 1;
+  bool use_cpu_weight = 2;
```

Comment:
lol proto3 sucks i guess?

it's a little hacky but you could just ignore stuff < 0 and remove all the bools in here. I think having them is gonna lead to fairly annoying code to generate this proto and read from it

Reply:
We need to distinguish between all of these cases:
(1) The field being unset. In this case, the executor should not set the cgroup value and leave it at the default. I think this behavior makes sense because of (a) old executor clients and (b) potential future additions to cgroup2
(2) Set to "max" value - we might want to explicitly set a value to be "unlimited." Unclear whether we'll want to do this in practice, but it feels like we're losing flexibility if we don't allow this to be represented.
(3) Explicitly set to 0
(4) Explicitly set to a value > 0

- Ignoring stuff < 0 doesn't allow for case (2) and also doesn't let us distinguish between (1) and (3).
- Using proto3 optional is fine but it means that the code will read like `if settings.CpuWeight != nil { doStuffWith(*settings.CpuWeight) }` I've gotten comments on PRs in the past that pointers to primitive types feel weird/unconventional because they require awkward code like this
 
An alternative would be to use a wrapper proto that's basically `oneof { max, int value, float value }` but this seems a bit much, idk.

Reply:
Don't optional fields generate GetCpuWeights and HasCpuWeights methods?

Reply:
it only generates `Get*` not `Has*`:

the generated field is

```
	CpuWeight                *int64                        `protobuf:"varint,1,opt,name=cpu_weight,json=cpuWeight,proto3,oneof" json:"cpu_weight,omitempty"`
```

and only this method is generated:

```
func (x *CgroupSettings) GetCpuWeight() int64 {
	if x != nil && x.CpuWeight != nil {
		return *x.CpuWeight
	}
	return 0
}
```

Reply:
(I'm fine with proto3 optional + `nil` checks if you both are cool with it? since it lets us get rid of these `use_*` fields)

Reply:
proto3 optional + nil checks seems good to me. It doesn't seem any harder to use than bool+field.

Reply:
I switched to a string representation per an offline discussion w/ Tyler. It allows representing unset, 0, non-zero, and "max" - the max value being represented as -1 with the optional approach has the downside of potentially storing it in the DB and messing up metrics.